### PR TITLE
TRUNK-5482 : Handled duplicates for Util method

### PR DESCRIPTION
**Pull request by LuGO0**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5482

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by LuGO0** [comment]
@sherrif10 @tendomart please review.

**Review by tendomart — COMMENTED** [review]
Seen

**Comment by LuGO0** [comment]
> Seen

Can we merge it and close the Issue? cc @tendomart @sherrif10 .

File: api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java [hunk 1]
```diff
@@ -1002,4 +1005,18 @@ public void applyLogLevels_shouldWarnWhenCalledWithInvalidLevel() {
...
+
+	/**
+	 * @see OpenmrsUtil#conceptListHelper(String)
+	 */
+	@Test
+	public void conceptListHelper_shouldNotReturnDuplicateConcepts() {
+		String descriptor = "name:YES | name:NO | name:YES";
+		List<Concept> ret = OpenmrsUtil.conceptListHelper(descriptor);
+		assertEquals(2, ret.size());
+
```
Comment (HerbertYiga) [inline api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java:1017]: @LuGO0  you can remove this space

File: api/src/main/java/org/openmrs/util/OpenmrsUtil.java [hunk 2]
```diff
@@ -855,12 +855,11 @@ public static int compareWithNullAsLatest(Date d1, Date d2) {
 		
 		return ret;
 	}
-	
-	// TODO: properly handle duplicates
+
```
Comment (HerbertYiga) [inline api/src/main/java/org/openmrs/util/OpenmrsUtil.java:858]: we also don't  need this space

File: api/src/main/java/org/openmrs/util/OpenmrsUtil.java [hunk 3]
```diff
@@ -855,12 +855,11 @@ public static int compareWithNullAsLatest(Date d1, Date d2) {
...
 	public static List<Concept> conceptListHelper(String descriptor) {
-		List<Concept> ret = new ArrayList<>();
+		Set<Concept> ret = new HashSet<>();
 		if (descriptor == null || descriptor.length() == 0) {
-			return ret;
+			return new ArrayList<>();
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/util/OpenmrsUtil.java:862]:
```suggestion
			return Collections.emptyList();
```

File: api/src/main/java/org/openmrs/util/OpenmrsUtil.java (hunk 2 above)
Comment (LuGO0) [reply api/src/main/java/org/openmrs/util/OpenmrsUtil.java:858]: will update @HerbertYiga

File: api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java (hunk 1 above)
Comment (LuGO0) [reply api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java:1017]: sure will update it @HerbertYiga

**Review by ibacher — APPROVED** [review]

**Pull request merged**